mitosheet/mitosheet/tests/updates/test_existing_import.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+"""
+Contains tests for existing import update events.
+"""
+import pandas as pd
+import os
+from mitosheet.tests.test_utils import create_mito_wrapper, create_mito_wrapper_dfs
+from mitosheet.tests.decorators import pandas_post_1_only, python_post_3_6_only
+
+TEST_EXCEL_FILE = 'excel_file.xlsx'
+TEST_CSV_FILE = 'csv_file.csv'
+TEST_CSV_FILE_TWO = 'csv_file_two.csv'
+
+
+@pandas_post_1_only
+@python_post_3_6_only
+def test_overwrite_multiple_imports():
+    # Make dataframes and files for test
+    df = pd.DataFrame(data={'A': [1, 2, 3], 'B': [2, 3, 4]})
+    df1 = pd.DataFrame(data={'C': [1, 2, 3], 'D': [2, 3, 4]})
+    df2 = pd.DataFrame(data={'E': [1, 2, 3], 'F': [2, 3, 4]})
+    df1.to_csv(TEST_CSV_FILE, index=False, sep=',', encoding='utf-8')
+    with pd.ExcelWriter(TEST_EXCEL_FILE) as writer:  
+        df.to_excel(writer, sheet_name='Sheet1', index=False)
+        df.to_excel(writer, sheet_name='Sheet2', index=False)
+        df.to_excel(writer, sheet_name='Sheet3', index=False)
+
+    
+    # Create with no dataframes
+    mito = create_mito_wrapper_dfs()
+    # And then import three sheets from the excel file
+    mito.excel_import(TEST_EXCEL_FILE, ['Sheet1', 'Sheet2', 'Sheet3'], True, 0)
+
+    step_id = mito.curr_step.step_id
+
+    # Update the imports 
+    updated_import_obj = [
+        {
+            'step_id': step_id,
+            'type': 'csv',
+            'import_params': {
+                'file_names': [TEST_CSV_FILE],
+                'delimeters': [','],
+                'encodings': ['utf-8'],
+                'error_bad_lines': [False],
+            }
+        },
+        {
+            'step_id': step_id,
+            'type': 'excel',
+            'import_params': {
+                'file_name': TEST_EXCEL_FILE,
+                'sheet_names': ['Sheet3'],
+                'has_headers': True,
+                'skiprows': 0,
+            }
+        },  
+        {
+            'step_id': step_id,
```

Comment:
Why do all of these have the same step id? Shouldn't they have the specific step id of the step they are trying to overwrite?

Reply:
Oh. I see. These are updating the same step that imports three import. This is then what leads to the complexity about needing to check how many times a specific step has been updated. 

Architecturally, I think you're right we want a single object that corresponds to each step. So it should look like:
```
{
             'step_id': step_id,
             'import_updates': [
                 {
                    'type': 'excel',
                     'import_params': {
                          'file_name': TEST_EXCEL_FILE,
                          'sheet_names': ['Sheet3'],
                          'has_headers': True,
                          'skiprows': 0,
                      },
                  },
                   {
                    'type': 'excel',
                     'import_params': {
                          'file_name': TEST_EXCEL_FILE,
                          'sheet_names': ['Sheet3'],
                          'has_headers': True,
                          'skiprows': 0,
                      },
                  },
                  ...
             ]
         }, 
```

Then, we insist on the variant that the number of import_updates is equal to the number of imports in that original step. 

This would simplify a ton of update logic, without adding much work at all!

Reply:
Yeah there is a tradeoff between having simple logic on the backend and having simpler logic on the frontend. If we move to this, then we'll have to keep track of more information about which specific sheet within which step is getting updated. 

We'll also still need to keep track of how many times we've updated that step on the backend so we can insert the steps in the correct location. 

I think there are other ways we could write the algorithm that would make it so we don't need to do that, but I'm not sure that this data type is actually a requirement for that. Although maybe we want it anyway